lib/utils/Utils.cpp

```diff
@@ -125,10 +125,15 @@ namespace MAT_NS_BEGIN {
     std::string GetTempDirectory()
     {
 #ifdef _WIN32
+        auto lpGetTempPathW = reinterpret_cast<decltype(&::GetTempPathW)>(GetProcAddress(GetModuleHandle(TEXT("kernel32")), "GetTempPath2W"));
```

Comment:
Could this be cached to avoid calling `GetProcAddress` every time to call `GetTempDirectory`?

Reply:
Never mind, saw the below section.

> Apps should call [GetTempPath2](https://learn.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-gettemppath2w) instead of GetTempPath.

Reply:
Since this is related to security, it just feels better to keep this fully dynamic. Presumably this is not a super common method/call to cause performance issues, right?

Reply:
The current usage of `GetTempDirectory` is confined to the SDK initialization phase, and I believe the additional `GetProcAddress` call in this context is manageable.